src/frontend/src/lib/services/address.services.ts

```diff
@@ -39,19 +55,51 @@ export const loadAddress = async (): Promise<{ success: boolean }> => {
 	return { success: true };
 };
 
-const saveEthAddressForFutureSignIn = async ({
+const loadBtcAddress = async ({
+	tokenId,
+	network
+}: {
+	tokenId: TokenId;
+	network: BitcoinNetwork;
+}): Promise<{ success: boolean }> =>
+	loadTokenAddress<BtcAddress>({
+		tokenId,
+		getAddress: (identity) =>
+			getBtcAddress({
+				identity,
+				network: network === 'mainnet' ? { mainnet: null } : { testnet: null }
+			}),
+		setIdbAddress: setIdbBtcAddressMainnet
+	});
+
+export const loadBtcAddressMainnet = async (): Promise<{ success: boolean }> =>
+	loadBtcAddress({
+		tokenId: BTC_MAINNET_TOKEN_ID,
+		network: 'mainnet'
+	});
+
+export const loadEthAddress = async (): Promise<{ success: boolean }> =>
+	loadTokenAddress<EthAddress>({
+		tokenId: ETHEREUM_TOKEN_ID,
+		getAddress: getEthAddress,
+		setIdbAddress: setIdbEthAddress
+	});
+
+const saveTokenAddressForFutureSignIn = async <T extends Address>({
 	identity,
-	address
+	address,
+	setIdbAddress
 }: {
 	identity: OptionIdentity;
-	address: EthAddress;
+	address: T;
+	setIdbAddress: (params: { address: IdbAddress<T>; principal: Principal }) => Promise<void>;
```

Comment:
Can the type of `params` be shortened to `SetIdbAddressParams` maybe?

Reply:
good catch! done